### PR TITLE
Reduce bitECS memory usage

### LINKED ISSUE #6291: Reduce memory usage
- Look for low-hanging fruit in memory profiling 
- Reduce and test lower ECS component sizes (from 100,000 to 10,000)
- #5954

### PR DESCRIPTION
**Pull request by speigg**
## Summary

Closes #6291 

## Checklist
- [ ] CI/CD checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Typescript passing via `npm run check-errors`
  - [ ] Unit & Integration tests passing via `npm run test`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

File: packages/engine/src/ecs/functions/ComponentFunctions.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import { ArrayByType, ISchema, Type } from 'bitecs'
 import { Engine } from '../classes/Engine'
 import { Entity } from '../classes/Entity'
 
+const INITIAL_COMPONENt_SIZE = 1000 // TODO set to 0 after next bitECS update
```
Comment (HexaField) [inline packages/engine/src/ecs/functions/ComponentFunctions.ts:8]: capitalization

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/ecs/functions/ComponentFunctions.ts (hunk 1 above)
Comment (speigg) [reply packages/engine/src/ecs/functions/ComponentFunctions.ts:8]: lol

Comment (HexaField) [reply packages/engine/src/ecs/functions/ComponentFunctions.ts:8]: lol

**Review by HexaField — APPROVED** [review]

**Comment by speigg** [comment]
Okay finally got those tests working, just waiting on CI/CD to verify
So, right now, bitECS's component resizing is totally borked
As soon as the entity count reaches 80% of the pre-specified capacity, it implodes
Just FYI
Should be fixed on next bitECS release 

Also, in order to get tests to pass w/ a default component size of 1000, I had to reset/delete the previous World every time we do createEngine() (otherwise, each test ends up accumulating the number of entities used in all components)

**Pull request merged**